src/components/Button/index.tsx

```diff
@@ -294,6 +311,85 @@ export function ButtonWithLink({ link, text, style }: { link: string; text: stri
   )
 }
 
+export function ButtonExternalLink({
+  link,
+  children,
+  disabled = false,
+}: {
+  link: string
+  disabled?: boolean
+  children: ReactNode
+}) {
+  const theme = useContext(ThemeContext)
+  return (
+    <ButtonPurpleDim disabled={disabled} as={disabled ? ButtonPurpleDim : ExternalLink} href={!disabled ? link : ''}>
+      {children}
+      <Box ml={2}>
+        <ArrowUpRight size="14px" color={theme.purple2} />
+      </Box>
+    </ButtonPurpleDim>
+  )
+}
+
+export function ButtonBadge({
+  to,
+  children,
+  number,
+  color = 'orange',
+  fit = false,
+  disabled = false,
+}: {
+  to: string
+  number: number
+  fit?: boolean
+  children: ReactNode
+  disabled?: boolean
+  color?: 'orange' | 'green' | 'red'
+}) {
+  return (
+    <ButtonPurpleDim fit={fit} as={disabled ? ButtonPurpleDim : Link} to={to} disabled={disabled}>
+      <Flex alignItems="center">
+        {children}
+        <Box ml={1}>
+          <NumberBadge badgeTheme={color}>{number}</NumberBadge>
+        </Box>
+      </Flex>
+    </ButtonPurpleDim>
+  )
+}
+
+export function ButtonLinkWithBadge({
+  link,
+  children,
+  number,
+  disabled = false,
+  color = 'orange',
+  fit = false,
+}: {
+  link: string
+  number: number
+  disabled?: boolean
+  fit?: boolean
+  children: ReactNode
+  color?: 'orange' | 'green' | 'red'
+}) {
+  return (
+    <ButtonPurpleDim
+      fit={fit}
+      disabled={disabled}
+      as={disabled ? ButtonPurpleDim : ExternalLink}
+      href={!disabled && link}
+    >
+      <Flex alignItems="center">
+        {children}
+        <Box ml={1}>
+          <NumberBadge badgeTheme={color}>{number}</NumberBadge>
+        </Box>
+      </Flex>
+    </ButtonPurpleDim>
+  )
+}
+
```

Comment:
We need to create some common patterns for buttons. I don't think this is entirely neccesary they are 90% similar and maybe should have been done where they are used

Reply:
This button, _ButtonLinkWithBadge_, is not being used. I thought I had deleted but might have been a merge that brought it back.

It's deleted now. We only have `ButtonBadge`.